Add "Remove" button on entity reference field widget

slashrsm opened the pull request:
https://www.drupal.org/node/2602552

slashrsm committed:
Issue #2602552 by slashrsm: Allow removal of entities from field widget.

primsi commented on src/Plugin/Field/FieldWidget/EntityReference.php:
```diff
@@ -209,8 +210,14 @@ function formElement(FieldItemListInterface $items, $delta, array $element, arra
     $ids = [];
-    if (($trigger = $form_state->getTriggeringElement()) && !empty($trigger['#ajax']['event']) && $trigger['#ajax']['event'] == 'entity_browser_value_updated' && in_array($this->fieldDefinition->getName(), $trigger['#parents'])) {
-      if ($value = $form_state->getValue($trigger['#parents'])) {
+    if (($trigger = $form_state->getTriggeringElement()) && in_array($this->fieldDefinition->getName(), $trigger['#parents'])) {
```
We could add a comment here, because it's not immediately clear why parents for this element aren't always the same.

janstoeckler committed:
Update EntityReference.php

In reference to issue #2602552 by slashrsm (https://github.com/slashrsm/entity_browser/commit/b5f60185dbb979625039c39ffa080a0a577b7012) we want to add form parent structure to fix validation fo nested form elements in accordance to \Drupal\Core\Form\FormValidator::handleErrorsWithLimitedValidation(). This should fix the removal of elements inside paragraph entity forms.

janstoeckler commented:
Added a pull request to make this work with nested form elements, e.g. with paragraphs. See https://github.com/slashrsm/entity_browser/pull/2.

slashrsm committed:
Merge pull request #2 from janstoeckler/patch-2

Update EntityReference.php
